src/openvic-simulation/modifier/ModifierManager.hpp

```diff
@@ -19,126 +16,136 @@ namespace OpenVic {
 		friend struct PopManager;
 		friend struct TechnologyManager;
 
-		using effect_variant_map_t = ordered_map<Modifier::modifier_type_t, ModifierEffect const*>;
+		using effect_variant_map_t = ordered_map<Modifier::modifier_origin_t, ModifierEffect const*>;
 
 		/* Some ModifierEffects are generated mid-load, such as max/min count modifiers for each building, so
 		 * we can't lock it until loading is over. This means we can't rely on locking for pointer stability,
 		 * so instead we store the effects in a deque which doesn't invalidate pointers on insert.
 		 */
 	private:
-		CaseInsensitiveIdentifierRegistry<ModifierEffect, RegistryStorageInfoDeque> IDENTIFIER_REGISTRY(modifier_effect);
+		CaseInsensitiveIdentifierRegistry<ModifierEffect, RegistryStorageInfoDeque> IDENTIFIER_REGISTRY(leader_modifier_effect);
```

Comment:
Define a type alias for this registry type and use it throughout this file and the corresponding cpp file.
A lot of duplicate code in this file can be avoided using macros.

Reply:
Good idea.